nsxt/metadata/metadata.go

```diff
@@ -0,0 +1,275 @@
+		if item.Metadata.SchemaType == "string" {
+			var value string
+			if len(parent) > 0 {
+				value = parentMap[key].(string)
+			} else {
+				value = d.Get(key).(string)
+			}
+			log.Printf("[INFO] assigning string %v to %s", value, key)
+			elem.FieldByName(item.Metadata.SdkFieldName).Set(reflect.ValueOf(&value))
+		}
+		if item.Metadata.SchemaType == "bool" {
+			var value bool
+			if len(parent) > 0 {
+				value = parentMap[key].(bool)
+			} else {
+				value = d.Get(key).(bool)
+			}
+			log.Printf("[INFO] assigning bool %v to %s", value, key)
+			elem.FieldByName(item.Metadata.SdkFieldName).Set(reflect.ValueOf(&value))
+		}
+		if item.Metadata.SchemaType == "int" {
+			var value int64
+			if len(parent) > 0 {
+				value = int64(parentMap[key].(int))
+			} else {
+				value = int64(d.Get(key).(int))
+			}
+			log.Printf("[INFO] assigning int %v to %s", value, key)
+			elem.FieldByName(item.Metadata.SdkFieldName).Set(reflect.ValueOf(&value))
+		}
+		if item.Metadata.SchemaType == "struct" {
+			nestedObj := reflect.New(item.Metadata.ReflectType)
+			var itemList []interface{}
+			if len(parent) > 0 {
+				itemList = parentMap[key].([]interface{})
+			} else {
+				itemList = d.Get(key).([]interface{})
+			}
+			if len(itemList) == 0 {
+				continue
+			}
+			nestedSchema := itemList[0].(map[string]interface{})
+
+			childElem := item.Schema.Elem.(*ExtendedResource)
+			SchemaToStruct(nestedObj.Elem(), d, childElem.Schema, key, nestedSchema)
+			log.Printf("[INFO] assigning struct %v to %s", nestedObj, key)
+			elem.FieldByName(item.Metadata.SdkFieldName).Set(nestedObj)
+		}
+		if item.Metadata.SchemaType == "list" || item.Metadata.SchemaType == "set" {
+			var itemList []interface{}
+			if item.Metadata.SchemaType == "list" {
+				if len(parent) > 0 {
+					itemList = parentMap[key].([]interface{})
+				} else {
+					itemList = d.Get(key).([]interface{})
+				}
+			} else {
+				if len(parent) > 0 {
+					itemList = parentMap[key].(*schema.Set).List()
+				} else {
+					itemList = d.Get(key).(*schema.Set).List()
+				}
+			}
+
+			if len(itemList) == 0 {
+				continue
+			}
+
+			// List of string, bool, int
+			if childElem, ok := item.Schema.Elem.(*ExtendedSchema); ok {
+				sliceElem := elem.FieldByName(item.Metadata.SdkFieldName)
+				switch childElem.Metadata.SchemaType {
+				case "string":
+					sliceElem.Set(
+						reflect.MakeSlice(reflect.TypeOf([]string{}), len(itemList), len(itemList)))
+				case "bool":
+					sliceElem.Set(
+						reflect.MakeSlice(reflect.TypeOf([]bool{}), len(itemList), len(itemList)))
+				case "int":
+					sliceElem.Set(
+						reflect.MakeSlice(reflect.TypeOf([]int64{}), len(itemList), len(itemList)))
+				}
+
+				for i, v := range itemList {
+					if childElem.Metadata.SchemaType == "int" {
+						sliceElem.Index(i).Set(reflect.ValueOf(v).Convert(reflect.TypeOf(int64(0))))
+					} else {
+						sliceElem.Index(i).Set(reflect.ValueOf(v))
+					}
+					log.Printf("[INFO] appending %v to %s", v, key)
```

Comment:
I think those printouts should be `TRACE` level, and we might want to preface them with some keyword so that the context of conversion is clear

Reply:
moved most to TRACE, except for the ones logged before returning an error.